src/lib/storage/up-storage.js

```diff
@@ -227,6 +227,57 @@ class ProxyStorage {
     headers[acceptEncoding] = headers[acceptEncoding] || 'gzip';
     // registry.npmjs.org will only return search result if user-agent include string 'npm'
     headers[userAgent] = headers[userAgent] || `npm (${this.userAgent})`;
+
+    if (!this.config.auth) {
+      return headers;
+    }
+
+    // copy headers to normalize keys
+    let copyHeaders = {};
+    Object.keys(headers).map((value) => {
+      copyHeaders[value.toLowerCase()] = headers[value];
+    });
+
+    // if header Authorization assigns this has precedence
+    if (copyHeaders['authorization']) {
+      return headers;
+    }
+
+    if (typeof this.config.auth !== 'object') {
+      throw new Error('Auth invalid');
+    }
+
+    let token = null;
+    // define authorization token to use in the Authorization Header
+    if (this.config.auth.token) {
+      token = this.config.auth.token;
+    } else if (this.config.auth.token_env) {
+      // get NPM_TOKEN http://blog.npmjs.org/post/118393368555/deploying-with-npm-private-modules
+      // or get other variable export in env
+      if (this.config.auth.token_env === true) {
```

Comment:
=== true ?

Reply:
To get value assigns in:
```shell
export NPM_TOKEN = "MY-jkdsjfksdjkfsdjfks"
```

Maybe is best, if assign auth.type get process.env.NPM_TOKEN and remove the value true.
```
if (!token) {
   token = process.env.NPM_TOKEN;
}
```

Reply:
Done